src/main/kotlin/org/rust/lang/core/resolve/ref/RsAttributeProcMacroReferenceImpl.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.resolve.ref
+
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementResolveResult
+import com.intellij.psi.ResolveResult
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsPath
+import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.resolve.collectResolveVariants
+import org.rust.lang.core.resolve.processAttributeProcMacroResolveVariants
+
+class RsAttributeProcMacroReferenceImpl(element: RsPath) : RsPathReferenceBase(element) {
```

Comment:
Why we can't use `RsReferenceCached` here? Looks like even with `referenceAnchor` and `advancedResolve` implemented twice it brings less copy-pasted code.

I made it differently with `RsMacroPathReferenceImpl` because I needed to cache a custom resolve result.

Reply:
Oh, thanks!
I've completely forgotten about it.
I've also made default implementation of `referenceAnchor` in `RsReferenceBase` so now `RsAttributeProcMacroReferenceImpl` impl is pretty minimalistic